client/my-sites/sharing/buttons/appearance.jsx

```diff
@@ -1,115 +1,169 @@
+import ButtonsStyle from './style';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackSite, isJetpackModuleActive } from 'state/sites/selectors';
+import { isPrivateSite } from 'state/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
 
-	getDefaultProps: function() {
-		return {
-			buttons: Object.freeze( [] ),
-			values: Object.freeze( {} ),
-			onChange: function() {},
-			onButtonsChange: function() {},
-			initialized: false,
-			saving: false
-		};
-	},
+class SharingButtonsAppearance extends Component {
+	static propTypes = {
+		buttons: PropTypes.array,
+		values: PropTypes.object,
+		onChange: PropTypes.func,
+		onButtonsChange: PropTypes.func,
+		onButtonsSave: PropTypes.func,
+		initialized: PropTypes.bool,
+		saving: PropTypes.bool,
+		isJetpack: PropTypes.bool,
+		isLikesModuleActive: PropTypes.bool,
+	}
 
-	onReblogsLikesCheckboxClicked: function( event ) {
-		this.props.onChange( event.target.name, ! event.target.checked );
+	static defaultProps = {
+		buttons: Object.freeze( [] ),
+		values: Object.freeze( {} ),
+		onChange: function() {},
+		onButtonsChange: function() {},
+		initialized: false,
+		saving: false
+	};
 
+	onReblogsLikesCheckboxClicked = event => {
+		this.props.onChange( event.target.name, ! event.target.checked );
 		if ( 'disabled_reblogs' === event.target.name ) {
-			analytics.ga.recordEvent( 'Sharing', 'Clicked Show Reblog Button Checkbox', 'checked', event.target.checked ? 1 : 0 );
+			this.props.recordGoogleEvent( 'Sharing', 'Clicked Show Reblog Button Checkbox', 'checked', event.target.checked ? 1 : 0 );
 		} else if ( 'disabled_likes' === event.target.name ) {
-			analytics.ga.recordEvent( 'Sharing', 'Clicked Show Like Button Checkbox', 'checked', event.target.checked ? 1 : 0 );
+			this.props.recordGoogleEvent( 'Sharing', 'Clicked Show Like Button Checkbox', 'checked', event.target.checked ? 1 : 0 );
 		}
-	},
+	};
 
-	getPreviewElement: function() {
+	getPreviewElement() {
 		if ( this.props.initialized ) {
+			const changeLabel = partial( this.props.onChange, 'sharing_label' );
+
 			return (
 				<ButtonsPreview
-					site={ this.props.site }
+					isPrivateSite={ this.props.isPrivate }
 					style={ this.props.values.sharing_button_style }
 					label={ this.props.values.sharing_label }
 					buttons={ this.props.buttons }
-					showLike={ ( ! this.props.site.jetpack || this.props.site.isModuleActive( 'likes' ) ) && ( '' === this.props.values.disabled_likes || false === this.props.values.disabled_likes ) }
-					showReblog={ ! this.props.site.jetpack && ( '' === this.props.values.disabled_reblogs || false === this.props.values.disabled_reblogs ) }
-					onLabelChange={ this.props.onChange.bind( null, 'sharing_label' ) }
+					showLike={
+						( ! this.props.isJetpack || this.props.isLikesModuleActive ) &&
+						( '' === this.props.values.disabled_likes || false === this.props.values.disabled_likes )
+					}
+					showReblog={
```

Comment:
I'm seeing the button on my website, so maybe they renamed it only for the reader?